### ISSUE TITLE
Warnings are surfaced during build when fragments with params are used in queries

### ISSUE THREAD

**Issue by brian-plummer-okcupid** [apollographql/apollo-kotlin#5311]
### Version

4.0.0-beta.1

### Summary

`failOnWarnings.set(true)` Build fails with the message:

`connectionqueries.graphql: (1, 23): Apollo: Variable "nextIntrosKey" is unused`

The query is defined as:

```
query IntrosPageQuery($nextIntrosKey: String) {
    me {
        ...ApolloNotificationCounts
        ...ApolloIntros
    }
}

```

The fragment is defined as:

```
fragment ApolloIntros on User {
    intros(after: $nextIntrosKey) {
        data {
            ...ApolloIntroUser
            ...ApolloFakeUser
        }
        pageInfo {
            ...ApolloPaging
        }
    }
}
```

It seems like this should work, if not please let me know what I can do to correct this. Thanks!

### Logs

```
Execution failed for task ':core-data:generateServiceApolloIr'.
> Apollo: Warnings found and 'failOnWarnings' is true, aborting.
```

**Comment by BoD** [comment]
I think this is fixed by https://github.com/apollographql/apollo-kotlin/pull/5290 but hasn't been released yet. In the meantime it can be tried out with the [snapshots](https://github.com/apollographql/apollo-kotlin#snapshots).

**Comment by brian-plummer-okcupid** [comment]
Thank you!

**Comment by BoD** [comment]
Closing as the fix is available in 4.0.0-beta.2
